### DISCUSSION TITLE
Project cannot start after upgrading from 3.3.3.110/120 to 3.10.0.296

### DISCUSSION THREAD

**Discussion by matthewtrahan** [MaxRev-Dev/gdal.netcore#168] [Q&A]
On our WPF project, everything works great when we're on MaxRev.Gdal.Core v3.3.3.120 and MaxRev.Gdal.WindowsRuntime.Minimal 3.3.3.110, but when we upgrade each of those to 3.10.0.296 and install MaxRev.Gdal.Universal, I get this error when trying to call GdalBase.ConfigureAll():

```
System.TypeInitializationException: The type initializer for 'OSGeo.GDAL.GdalPINVOKE' threw an exception.
 ---> System.TypeInitializationException: The type initializer for 'SWIGExceptionHelper' threw an exception.
 ---> System.DllNotFoundException: Unable to load DLL 'gdal_wrap' or one of its dependencies: The specified procedure could not be found. (0x8007007F)
   at OSGeo.GDAL.GdalPINVOKE.SWIGExceptionHelper.SWIGRegisterExceptionCallbacks_Gdal(ExceptionDelegate applicationDelegate, ExceptionDelegate arithmeticDelegate, ExceptionDelegate divideByZeroDelegate, ExceptionDelegate indexOutOfRangeDelegate, ExceptionDelegate invalidCastDelegate, ExceptionDelegate invalidOperationDelegate, ExceptionDelegate ioDelegate, ExceptionDelegate nullReferenceDelegate, ExceptionDelegate outOfMemoryDelegate, ExceptionDelegate overflowDelegate, ExceptionDelegate systemExceptionDelegate)
   at OSGeo.GDAL.GdalPINVOKE.SWIGExceptionHelper..cctor()
   --- End of inner exception stack trace ---
   at OSGeo.GDAL.GdalPINVOKE.SWIGExceptionHelper..ctor()
   at OSGeo.GDAL.GdalPINVOKE..cctor()
   --- End of inner exception stack trace ---
   at OSGeo.GDAL.GdalPINVOKE.AllRegister()
   at OSGeo.GDAL.Gdal.AllRegister()
   at MaxRev.Gdal.Core.GdalBase.ConfigureGdalDrivers(String gdalDataFolder)
   at MaxRev.Gdal.Core.GdalBase.ConfigureAll()
```

I tried a new solution and did the above and called GdalBase.ConfigureAll() and everything worked, so I'm not sure why it's not finding the runtimes in our application. I can verify that gdal_wrap and everything else is available in this folder (D:\source\App\App\App.Application\bin\x64\Debug\runtimes\win-x64\native) pre and post-upgrade so I'm not sure what I'm missing.

Thank you for the help!

**Comment by MaxRev-Dev** [comment]
@matthewtrahan Can you create a MRE and attach it here?
Let me clarify, does it work on in a fresh/empty project? Then have you tried to cleanup build artifacts (bin/obj directories)?

**Reply by matthewtrahan** [reply]
Correct, it does work on a fresh/empty project. The one difference I see between the two is that my app has x64 in there between bin and Debug whereas the new solution does not. 

I have tried removing the bin and obj files, but I get the same error.

Is an MRE the output of ldd gdal.dll? If so, attached here.
[output.txt](https://github.com/user-attachments/files/17720428/output.txt)

**Reply by MaxRev-Dev** [reply]
Sorry for confusion, MRE is minimal reproducible example.
I wanted to see a difference between current project setup (the one that fails) and an empty one.

So if the "x64" part differs - please, double check the solution configuration. 

Also, did you modify the **working** directory in csproj?

```xml
<PropertyGroup>
    <RunWorkingDirectory>$(WorkFolder)</RunWorkingDirectory>
</PropertyGroup>
```

**Reply by matthewtrahan** [reply]
Hmm, yeah, even after making the new solution have the x64 in the output path, it still works.

I added the `RunWorkingDirectory` property to my csproj, but it still fails in the same way. Do I need to set that anywhere or does MSBuild do it? I'll keep messing with the potential MRE and see if I can reproduce over there. Any other ideas are greatly appreciated
